### ISSUE TITLE
Nondeterministic results

### ISSUE THREAD

**Issue by nerdyrodent** [logtd/ComfyUI-Fluxtapoz#10]
Hello!
I do love a bit of unsampling, however when using example_rf_inversion_updated.json I'm getting non-deterministic results:

* Run the worfklow, open the resulting image in a new tab / save to compare
* Change the prompt (e.g. insert a full-stop, or change guidance to 0.1) for the forward sampler, generate and ignore
* Revert the change you just made, generate and compare to the first image

This is with the fixed seed of 0, but it seems any change to the conditioning nets a random result?

**Comment by logtd** [comment]
Hey, thanks for the report. 

Is this happening with the most recent version with the updated workflow and the seed set to fixed?

**Comment by nerdyrodent** [comment]
Yes, using the latest update and a fixed seed of 0 with example_rf_inversion_updated.json still does the same thing.

**Comment by logtd** [comment]
I was able to reproduce these images after changing params back and forth, and also turning comfy on and off.

![ComfyUI_temp_bxccf_00001_](https://github.com/user-attachments/assets/44a0468b-4d5a-46d1-a678-83e1c46f41b0)

Are you seeing drastic differences or tiny changes?

And just to confirm the following seed is set to fixed: 

<img width="354" alt="Screenshot 2024-10-22 at 2 09 11 PM" src="https://github.com/user-attachments/assets/18fe5854-fabb-47b9-b13d-b67d84cc50c6">

**Comment by nerdyrodent** [comment]
It's different every time for me, though can't say why :|
![Screenshot from 2024-10-23 09-23-57](https://github.com/user-attachments/assets/94cdbcf8-aff4-490c-b5ac-80d5a5f28a18)

**Comment by logtd** [comment]
Ohhh. I see what's happening. Let me take a look at that later

**Comment by nerdyrodent** [comment]
The bottom is just a way to see if I could use the node caching to "keep" where it went

**Comment by logtd** [comment]
I'll convert this to use Comfy's noise generation in the future to ensure seeds across the environment are respected -- thanks for reporting this

**Comment by dataleveling** [comment]
I faced the same issue when I was testing Guidance vs DeGuidance.

Change unsampling and sampling's Guidance node:
1. Connect to Guidance
2. Connect to DeGuidance
3. Connect back to Guidance
4. Connect back to DeGuidance
5. Results from 1 and 3 are different, 2 and 4 are different.
![Screenshot 2024-12-09 151228](https://github.com/user-attachments/assets/37fe89dc-b1ff-4c75-b6ab-91a5545a6a96)

Keep unsampling and only change sampling's Guidance node:
1. Connect to Guidance
2. Connect to DeGuidance
3. Connect back to Guidance
4. Connect back to DeGuidance
5. Results from 1, 2, 3 and 4 are same.
![Screenshot 2024-12-09 151208](https://github.com/user-attachments/assets/cf88fc4d-7b23-4d9b-95a8-691106fbaea2)

My guess is the result from the unsampling steps is not an identical result with each run despite same parameters if there are minor changes or reconnecting of nodes in the unsampling conditioning portion.

My main concern is the reproducibility of image from a workflow.
Results from before / after shutting down comfyui server and rerun workflow:
![ComfyUI_temp_tsnod_00001_](https://github.com/user-attachments/assets/a3fef486-24db-4ef4-993b-d30ede558a52)
![ComfyUI_temp_tcfnp_00001_](https://github.com/user-attachments/assets/217645d7-5337-4834-887d-6e6fd55449f7)

**Comment by logtd** [comment]
Hey folks, great information here.

Yes, the issue is in the noise injection that happens in the unsampling sampler. I'll fix this soon
